ocaml/lambda/lambda.ml

```diff
@@ -356,6 +356,15 @@ let must_be_value layout =
       Pgenval
   | _ -> Misc.fatal_error "Layout is not a value"
 
+(* [layout_top] is unused and will be removed.  [layout_bottom] is for things
+   that don't return - we can pick whatever sort. *)
```

Comment:
The result for `layout_bottom` seems a bit dangerous. If something does not return, it can still be inside an `if` statement with like `if f () then failwith "" else #0.`, and then we have an if-then-else where both branches return a different sort. What would happen if we added the equivalent of `layout_bottom` as the bottom element of the sort lattice instead?

Reply:
I agree that seems weird / scary.

We can't really add bottom to the type system.  It's important that layouts with different sorts have no intersection for the way inference works now.

I can think of a couple alternatives:
1) Change `Pbottom` to carry a type system sort.  Though at a quick glance it looks like it might be tricky to get this information to the places we create `Pbottom`.
2) Just delete `sort_of_layout` and pass a sort down to the places where this is called.  This would make me a tiny bit sad, but is probably easiest.

Reply:
I've pushed a commit that eliminates this function with option (2).

I hesitated to go with option (1) mainly because it feels wrong to store type system sort information in lambda - we have otherwise dropped or translated all type system stuff at this point.  Maybe a constant sort isn't so bad, or a new type that's just a simplified `Lambda.layout` without the `value_kind` or the top and bottom cases.

Option (2) required a fair bit of plumbing and many functions now take both sorts and layouts, which feels a little inelegant.  So I am happy to try one of those approaches for option (1) if you two think it would be better than what I just pushed.